File: macos/Scanner/Sources/Scanner/Rules/AESAUTH004.swift
public struct AESAUTH004: Rule {
    public let id = "AES-AUTH-004"
    public let severity: Severity = .critical

    public init() {}

    public func evaluate(_ facts: ScanFacts) -> [Finding] {
        facts.mcpServers.compactMap { server in
            guard !server.disabled else {
                return nil
            }
            let match = server.env.first { key, value in
                AiThreatPatterns.isCloudCredentialEnv(key: key, value: value)
            }
            guard let match else {
                return nil
            }
            return finding(
                app: server.appId,
                serverName: server.name,
                affectedPath: server.configPath,
                maskedValue: SecretPatterns.masked(match.value)
            )
        }
    }
}

File: windows/Scanner/Rules/AESAUTH004.cs
using AIExposureScanner.Scanner.Models;

namespace AIExposureScanner.Scanner.Rules;

public sealed class AESAUTH004 : IRule
{
    public string Id => "AES-AUTH-004";
    public Severity Severity => Severity.Critical;

    public IReadOnlyList<Finding> Evaluate(ScanFacts facts)
    {
        var findings = new List<Finding>();
        foreach (var server in facts.McpServers.Where(server => !server.Disabled))
        {
            var match = server.Env.FirstOrDefault(kv => AiThreatPatterns.IsCloudCredentialEnv(kv.Key, kv.Value));
            if (match.Key is null)
            {
                continue;
            }
            findings.Add(this.Finding(
                server.AppId,
                server.Name,
                affectedPath: server.ConfigPath,
                maskedValue: SecretPatterns.Masked(match.Value)
            ));
        }
        return findings;
    }
}
